dvc/utils/threadpool.py

```diff
@@ -29,6 +36,12 @@ def create_taskset(n: int) -> Set[futures.Future]:
             done, tasks = futures.wait(
                 tasks, return_when=futures.FIRST_COMPLETED
             )
-            for fut in done:
-                yield fut.result()
+            try:
+                for fut in done:
+                    yield fut.result()
+            except Exception:
+                if cancel_futures:
+                    for fut in tasks:
+                        fut.cancel()
```

Comment:
@pmrowla, we don't need this if we backport the `cancel_futures=True` support to `Python<3.9`, right? 

We may need to extend `Executor.__exit__` to send `cancel_futures=True` as well. Let me try to propose something, hmm.

Reply:
Right

Reply:
Proposed in #7602.